realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ProcessCommitTests.java

```diff
@@ -16,159 +16,202 @@
-                                }
-                            })
-                            .build();
-                    Realm.deleteRealm(syncConfig);//TODO do this in Rule as async tests
-                    final Realm realm = Realm.getInstance(syncConfig);
-                    Intent intent = new Intent(targetContext, SendOneCommit.class);
-                    targetContext.startService(intent);
-                    final RealmResults<ProcessInfo> all = realm.where(ProcessInfo.class).findAll();
-                    all.addChangeListener(new RealmChangeListener<RealmResults<ProcessInfo>>() {
-                        @Override
-                        public void onChange(RealmResults<ProcessInfo> element) {
-                            assertEquals(1, all.size());
-                            assertEquals("Background_Process1", all.get(0).getName());
-                            testFinished.countDown();
-                        }
-                    });
-
-                    Looper.loop();
-
-                } catch (Throwable e) {
-                    exception[0] = e;
-                    testFinished.countDown();
-                }
+            public void onChange(RealmResults<ProcessInfo> element) {
+                assertEquals(1, all.size());
+                assertEquals("Background_Process1", all.get(0).getName());
+                realm.close();
+                user.logout();
+
+                remoteService.triggerServiceStep(SimpleCommitRemoteService.stepB_closeRealmAndLogOut);
+
+                looperThread.testComplete();
             }
         });
-        boolean testTimedOut = testFinished.await(300, TimeUnit.SECONDS);
-        if (exception[0] != null) {
-            throw exception[0];
-        } else if (!testTimedOut) {
-            fail("Test timed out ");
-        }
+
+        remoteService.triggerServiceStep(SimpleCommitRemoteService.stepA_openRealmAndCreateOneObject);
     }
 
-    // TODO:
-    // - send string from service and match replicate integration tests from Cocoa
-    // - add gradle task to start the sh script automatically (create pid file, ==> run or kill existing process)
-    // - check the requirement for the issue again
+    public static class ALotCommitsRemoteService extends RemoteIntegrationTestService {
+        private static SyncUser user;
+        public static final Step stepA_openRealm = new Step(RemoteTestService.BASE_A_LOT_COMMITS, 1) {
+
+            @Override
+            protected void run() {
+                user = UserFactory.getInstance().loginWithDefaultUser(Constants.AUTH_URL);
+                String realmUrl = Constants.SYNC_SERVER_URL;
+
+                final SyncConfiguration syncConfig = new SyncConfiguration.Builder(user, realmUrl)
+                        .directory(getService().getRoot())
+                        .name(UUID.randomUUID().toString() + ".realm")
+                        .build();
+                getService().realm = Realm.getInstance(syncConfig);
```

Comment:
I think @bmeike would prefer if this was `setRealm()` and getRealm()` instead :)